src/ui/map.ts

```diff
@@ -121,15 +138,59 @@ class MapUI {
         }
       }
     };
+
+    const onStyleLoad = async () => {
+      MapUI.customMarkerRawSvg = undefined;
+      const style = map.getStyle();
+
+      const customMarkers = (style.metadata as any)?.['radar:customMarkers'];
+      if (Array.isArray(customMarkers) && customMarkers.length > 0) {
+        const customMarker = customMarkers[0]; // only support one custom marker for now
+        try {
+          const markerRawSvg = await Http.request({
+            method: 'GET',
+            versioned: false,
+            path: `maps/markers/${customMarker.id}`,
+            headers: {
+              'Content-Type': 'image/svg+xml',
+            },
+          });
+          MapUI.customMarkerRawSvg = markerRawSvg.data;
+        } catch (err) {
+          Logger.warn(`Error getting custom marker: ${customMarker.id} - using default marker.`);
+        }
+      }
+
+      // set markers if necessary
```

Comment:
This is to handle dynamic style changes? One thing to keep in mind, we should only do this if it is already using a default marker.